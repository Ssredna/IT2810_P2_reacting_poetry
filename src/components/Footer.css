.Footer-container {
  color: white;
  height: min-content;
  width: 100vw;
  display: flex;
  align-items: center;
  flex-direction: column;
  text-align: center;
}

.Footer-header {
  margin-top: 1rem;
  margin-bottom: 0;
}

.Footer-text-top {
  margin-top: 1rem;
  margin-bottom: 0;
}

.Footer-text-bottom {
  margin-top: 0;
  margin-bottom: 1rem;
}

.Footer-link:visited,
.Footer-link:hover,
.Footer-link:link,
.Footer-link:active {
  color: white;
}
