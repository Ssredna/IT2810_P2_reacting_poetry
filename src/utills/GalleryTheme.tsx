export const ColorTheme: AllColorThemesInterface = {
  theme1: {
    color1: '#fcc5c5',
    color2: '#D4AD9B',
    complementary: '#c0ffd5',
    name: {
      english: 'Pink',
      norwegian: 'Rosa',
    },
  },
  theme2: {
    color1: '#BBEDD7',
    color2: '#7fddb3',
    complementary: '#edbbd1',
    name: {
      english: 'Green',
      norwegian: 'Grønn',
    },
  },
  theme3: {
    color1: '#add8e6',
    color2: '#72bcd4',
    complementary: '#e6d8ad',
    name: {
      english: 'Blue',
      norwegian: 'Blå',
    },
  },
};

interface AllColorThemesInterface {
  theme1: ColorThemeInterface;
  theme2: ColorThemeInterface;
  theme3: ColorThemeInterface;
}

export interface ColorThemeInterface {
  color1: string;
  color2: string;
  complementary: string;
  figureColor?: string;
  name: languageOptions;
}

export interface languageOptions {
  english: string;
  norwegian: string;
}

export enum Languages {
  EN = 'english',
  NO = 'norwegian',
}
