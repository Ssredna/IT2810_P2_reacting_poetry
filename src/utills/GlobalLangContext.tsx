import React from 'react';

/* Oppretter en Context for Global Language */
const GlobalLangContext = React.createContext('');
/* Eksporterer Context Provider som den som eier context skal bruke */
export const GlobalLangProvider = GlobalLangContext.Provider;

/* Eksporterer context som forbrukerne bruker med useContext
-hooken */
export default GlobalLangContext;

/* Translations const!  */
export const translations = {
  english: {
    gallery: 'Gallery',
    home: 'Home',
    theme: 'Theme',
    figAmount: 'Number of figures',
    fancyMode: 'Fancymode',
    off: 'off',
    on: 'on',
    close: 'Close',
    createdBy: 'Created by',
    project: 'project',
    welcomeText: 'Welcome to',
    toGallery: 'To gallery',
    noFavorites: 'You do not have any favorites yet!',
    galleryLoading: 'Gallery is loading...',
  },
  norwegian: {
    gallery: 'Galleri',
    home: 'Hjem',
    theme: 'Tema',
    figAmount: 'Antall figurer',
    fancyMode: 'Fancymodus',
    off: 'av',
    on: 'p√•',
    close: 'lukk',
    createdBy: 'Laget av',
    project: 'prosjekt',
    welcomeText: 'Velkommen til',
    toGallery: 'Til galleriet',
    noFavorites: 'Du har ikke lagt til noen favoritter enda!',
    galleryLoading: 'Laster inn galleri...',
  },
};
