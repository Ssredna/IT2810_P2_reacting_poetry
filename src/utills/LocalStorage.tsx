import { useState } from 'react';

export enum StorageType {
  LocalStorage = 'localStorage',
  SessionStorage = 'sessionStorage',
}

export const useWebStorage = (key, initialValue, storageType: StorageType) => {
  const [storedValue, setStoredValue] = useState(() => {
    try {
      const item = window[storageType].getItem(key);
      return item ? JSON.parse(item) : initialValue;
    } catch (err) {
      console.error(err);
      return initialValue;
    }
  });

  const setValue = (value) => {
    try {
      const valueToStore =
        value instanceof Function ? value(storedValue) : value;
      setStoredValue(valueToStore);
      window[storageType].setItem(key, JSON.stringify(valueToStore));
    } catch (err) {
      console.error(err);
    }
  };

  return [storedValue, setValue];
};

export default useWebStorage;
