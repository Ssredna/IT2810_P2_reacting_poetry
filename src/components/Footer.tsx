import React from 'react';
import './Footer.css';
import '../ColorPalette.css';
import GlobalLangContext, { translations } from '../utills/GlobalLangContext';

export default class Footer extends React.Component {
  static contextType = GlobalLangContext;

  render() {
    return (
      <div className="Footer-container bg-primary-1">
        <h2 className="Footer-header">
          IT2810 Webutvikling {translations[this.context].project} 2
        </h2>
        <p className="Footer-text-top">
          {translations[this.context].createdBy} Balluni & Anesh
        </p>
        <p className="Footer-text-bottom">
          <a
            href="https://gitlab.stud.idi.ntnu.no/it2810-h20/team-58/reacting-poetry"
            className="Footer-link"
          >
            GitLab repository
          </a>
        </p>
      </div>
    );
  }
}
