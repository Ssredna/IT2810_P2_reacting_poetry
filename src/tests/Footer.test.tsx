import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../components/Footer';
import GlobalLangContext from '../utills/GlobalLangContext';
import { Languages } from '../utills/GalleryTheme';

test('it renders correctly', () => {
  const tree = renderer
    .create(
      <GlobalLangContext.Provider value={Languages.EN}>
        <Footer />
      </GlobalLangContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
