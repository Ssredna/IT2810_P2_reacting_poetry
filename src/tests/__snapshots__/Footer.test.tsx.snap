// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="Footer-container bg-primary-1"
>
  <h2
    className="Footer-header"
  >
    IT2810 Webutvikling 
    project
     2
  </h2>
  <p
    className="Footer-text-top"
  >
    Created by
     Balluni & Anesh
  </p>
  <p
    className="Footer-text-bottom"
  >
    <a
      className="Footer-link"
      href="https://gitlab.stud.idi.ntnu.no/it2810-h20/team-58/reacting-poetry"
    >
      GitLab repository
    </a>
  </p>
</div>
`;
